### PR TITLE
add nullability annotations to lib/web_ui painting.dart

### PR DESCRIPTION
**Pull request by yjbanov**
The web version of https://github.com/flutter/engine/pull/18374

## Related Issues

https://github.com/flutter/flutter/issues/53661

### CODE REVIEW

File: lib/web_ui/lib/src/ui/painting.dart [hunk 1]
```diff
@@ -1750,7 +1758,7 @@ class Shadow {
   /// Values for `t` are usually obtained from an [Animation<double>], such as
   /// an [AnimationController].
   /// {@endtemplate}
-  static Shadow lerp(Shadow a, Shadow b, double t) {
+  static Shadow/*!*/ lerp(Shadow/*?*/ a, Shadow/*?*/ b, double/*!*/ t) {
```
Comment (zanderso) [inline lib/web_ui/lib/src/ui/painting.dart:1761]: Looks like this can return null.

Comment (yjbanov) [reply lib/web_ui/lib/src/ui/painting.dart:1761]: Done.

**Comment by yjbanov** [comment]
PTAL.

File: lib/web_ui/lib/src/ui/painting.dart [hunk 2]
```diff
@@ -1601,13 +1604,13 @@ String _instantiateImageCodec(
   return null;
 }
 
-Future<Codec> webOnlyInstantiateImageCodecFromUrl(Uri uri,
-    {engine.WebOnlyImageCodecChunkCallback chunkCallback}) {
+Future<Codec> webOnlyInstantiateImageCodecFromUrl(Uri/*!*/ uri,
```
Comment (zanderso) [inline lib/web_ui/lib/src/ui/painting.dart:1607]:
```suggestion
Future<Codec/*?*/> webOnlyInstantiateImageCodecFromUrl(Uri/*!*/ uri,
```

Comment (yjbanov) [reply lib/web_ui/lib/src/ui/painting.dart]: Done.

**Comment by yjbanov** [comment]
The luci-engine failure is due to https://github.com/flutter/flutter/issues/58785. Ignoring it.

**Pull request merged**